libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -250,13 +257,20 @@ void AP_ICEngine::update(void)
 
 /*
   check for throttle override. This allows the ICE controller to force
-  the correct starting throttle when starting the engine
+  the correct starting throttle when starting the engine and maintain idle when disarmed
  */
 bool AP_ICEngine::throttle_override(uint8_t &percentage)
 {
     if (!enable) {
         return false;
     }
+
+    uint8_t current_throttle = SRV_Channels::get_output_scaled(SRV_Channel::k_throttle);
+    if (idle_percent > current_throttle && state == ICE_RUNNING) {
+        percentage = (uint8_t)idle_percent;
+        return true;
+    }
```

Comment:
Why is this looking at SRV_Channel::k_throttle? It should be looking at the demanded throttle "percentage" instead.

if (state == ICE_RUNNING && idle_percent > 0 && idle_percent > percentage) {
    percentage = (uint8_t)idle_percent;
    return true;
}

Reply:
i'm not sure I follow, I should set the percentage to current throttle here? 
https://github.com/ArduPilot/ardupilot/blob/c6115f4ff36b3e2f47172d5feb10755200dc9f46/ArduPlane/servos.cpp#L790

Reply:
oh, yeah. Sorry, got it mixed up with a branch where I've totally re-written ICE. This is fine as-is except get)output_scaled returns a signed value so that needs to be handled. Also, mind moving the "state == " to the front so it's more obvious when it happens?

Reply:
I pushed a change to your branch